File: osr_ground/kml/pipe_send.py
#!/usr/bin/python

import time
import sys
import os, pipes

if not os.path.exists('/tmp/rocket_pipe'):
    os.mkfifo('/tmp/rocket_pipe')
    print 'Created pipe for write'

pipeout = os.open('/tmp/rocket_pipe', os.O_WRONLY)
print 'Opened pipe for write'
coords = open('coords', 'r')

index = 0
t = 0.0
#-10x^2+500x
while 1:
    
    lat = 44.565 + (0.0001 * index)
    lon = -123.278 + (0.0003 * index)
    alt = (-10 * t * t) + (500 * t)
    if alt < 0:
        alt = 0
    
    line = str(lat) + ',' + str(lon) + ',' + str(alt) + '\n'
    os.write(pipeout, line)
    
    if t > 0 and alt == 0:
        print 'end of path'
        break

    index += 1
    t += 0.25
    print 'sent: %s' % line
    time.sleep(0.25)

File: osr_ground/kml/pipe_send.c
#define _POSIX_SOURCE
#define _POSIX_C_SOURCE 200809L
#define _BSD_SOURCE

#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <errno.h>
#include <fcntl.h>
#include <sys/wait.h>
#include <sys/stat.h>
#include <sys/types.h>
#include <string.h>
#include <strings.h>
#include <ctype.h>

/*    */
int main()
{
	int pipeout = open("/tmp/rocket_pipe", O_WRONLY);

	int i;
	for (i = 0; i < 10; i++) {
		char* line;
		sprintf(line, "%d\n", i);
		printf("Send: %s", line);
		write(pipeout, line, strlen(line));
		sleep(1);
	}

	close(pipeout);
	return 0;
}
